Sources_v3/WebSocketClient/Engine/StarscreamWebSocketEngine.swift

```diff
@@ -46,19 +46,28 @@ import Foundation
     }
     
     extension StarscreamWebSocketProvider: Starscream.WebSocketDelegate {
-        func websocketDidConnect(socket: Starscream.WebSocketClient) {
-            delegate?.webSocketDidConnect()
+        func didReceive(event: WebSocketEvent, client: WebSocket) {
+            switch event {
+            case .connected:
+                delegate?.webSocketDidConnect()
+            case let .text(text):
+                delegate?.webSocketDidReceiveMessage(text)
+            case let .error(error):
+                delegate?.webSocketDidDisconnect(error: error.map(WebSocketEngineError.init))
+            case let .disconnected(reason, code):
+                let error = WebSocketEngineError(reason: reason, code: Int(code), engineError: nil)
+                delegate?.webSocketDidDisconnect(error: error)
+            case .cancelled:
+                let error = WebSocketEngineError(
+                    reason: "Cancelled",
+                    code: WebSocketEngineError.stopErrorCode,
```

Comment:
I'm not sure `WebSocketEngineError.stopErrorCode` suites here 🤔  We might want some other error code and handle it a different way

Reply:
Hmm, I wouldn't use stop code here 🤔 It's usually used by the backend signalizing that reconnecting is not suitable.
Just put `-1` there probably :)

Reply:
fixed for both v2 and v3